### DISCUSSION TITLE
Example for Distillation

### DISCUSSION THREAD

**Discussion by Bene94** [IDAES/idaes-pse#246] [Q&A]
Hello,

I have a student trying out the IDAES-Framework and we want to use a distillation in our flowsheet. I was wondering if there is a working example for the distillation column unit model and if there are any limitations of the distillation we should be aware of? 

Best,
Benedikt

**Comment by adowling2** [comment]
@jghouse88 Any thoughts?

One of my students, @agarciadiego, plans to try the distillation model during the next two months.

**Comment by jghouse88** [comment]
@Bene94  we do not have an example (like a jupyter notebook) in our examples repo yet but you can take a look at the tests to help understand setting up the distillation column: https://github.com/IDAES/idaes-pse/blob/main/idaes/generic_models/unit_models/distillation/tests/test_tray_column.py 

Regarding limitations, we have an initialization procedure that works on the tests cases we have tried. However, it might need custom initialization depending on your system and column conditions. What system do you plan to model?

**Reply by songkuku** [reply]
Hello. This is Ayoung who's working with @Bene94. At the moment, we just need to model a distillation column that processes different components other than benzene, toluene, and o-xylene as shown in the example. Therefore, I am revising a parameter block used for a test example (https://github.com/IDAES/idaes-pse/blob/main/idaes/generic_models/unit_models/distillation/tests/test_tray_column.py).
I could find most of the parameters, but it's difficult to find the parameter for ideal gas heat capacity which was referred to [1] The Properties of Gases and Liquids (1987), 4th edition, Chemical Engineering Series - Robert C. Reid. May I kindly ask whether you referred to a **table called 'Section C. Ideal Gas and Liquid Heat Capacities' at page A.35?** I couldn't obtained the ideal gas heat capacity parameters for benzene, toluene, and o-xylene although I tried with gas constant (8.314 J/mol-K). Thanks a lot!

Best,
Ayoung

**Reply by andrewlee94** [reply]
Hi @songkuku, I do not have the source book on hand at the moment, however these values did come from the appendix of the referenced text.

If you are having trouble finding parameters in the Properties of Gases and Liquids, you could look at using the generic properties framework, which supports a number of different options available for calculating `cp`, including the NIST Webbook which is publicly available and Perry's Handbook (7th edition). Examples (including a benzene-toluene example), can be found here:

https://github.com/IDAES/idaes-pse/tree/main/idaes/generic_models/properties/core/examples

**Comment by Bene94** [comment]
Hello,

Thank you for the previous help. We are, however, still struggling with the distillation. 

Currently, we are trying to get the distillation to run with ethanol and water(Ideal so no Azeo). However, the initialization of the distillation collum always crashes.  As property data, we modified the benzene and toluene example and used data from the same sources. 

Are there known issues with the initialization of the distillation? 

I attached the property data and the test flowsheet. I have the feeling it might be due to the water, as an ethanol-benzene system worked.


[Distillation.zip](https://github.com/IDAES/idaes-pse/files/6320681/Distillation.zip)

**Reply by jghouse88** [reply]
I checked out the zip files but the notebook is still using the benzene-toluene prop pack. Can you send me the file that is actually causing trouble i.e. the one using the ethanol-water prop pack? It will be easier to debug if I can have the same inputs you are using.

**Reply by Bene94** [reply]
Sorry, I did not save the file before zipping, now it should be the right one. 

Thank you for your fast response!

[test_dist.zip](https://github.com/IDAES/idaes-pse/files/6321226/test_dist.zip)

**Reply by jghouse88** [reply]
Thank you, I will take a look and get back.
